packages/overlay/src/App.tsx

```diff
@@ -136,10 +136,10 @@ export default function App({
   // See https://github.com/remix-run/react-router/issues/7634
   const navigate = useNavigate();
   const clearEvents = useCallback(async () => {
-    const { origin } = new URL(sidecarUrl);
-    const clearEventsUrl: string = `${origin}/clear`;
-
     try {
+      const { origin } = new URL(sidecarUrl);
+      const clearEventsUrl: string = `${origin}/clear`;
```

Comment:
Why move these into the `try-catch` block? Due to potentially malformed URL issues?

Reply:
yes correct